src/greentest/test__socket.py

```diff
@@ -313,6 +313,20 @@ class io(object):
     # Creating new types in the function takes a cycle to cleanup.
     test_wait_timeout.ignore_leakcheck = True
 
+    def test_connect_ex(self):
+        # Issue 841
+        cli = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.addCleanup(cli.close)
```

Comment:
The test harness counts these sockets as leaked:
```
  ======================================================================
  FAIL: test_connect_ex (__main__.TestFunctions)
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/home/travis/build/gevent/gevent/src/greentest/greentest.py", line 243, in wrapped
      raise AssertionError('refcount increased by %r\n%s' % (deltas, diff))
  AssertionError: refcount increased by [4, 4, 4]
  <type 'instancemethod'>: 62 != 63
  <type 'gevent.libev.corecext.io'>: 4 != 6
  <class 'gevent._socket2.socket'>: 2 != 3
  
  ======================================================================
  FAIL: test_connect_ex_gaierror (__main__.TestFunctions)
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/home/travis/build/gevent/gevent/src/greentest/greentest.py", line 243, in wrapped
      raise AssertionError('refcount increased by %r\n%s' % (deltas, diff))
  AssertionError: refcount increased by [10, 436, 4, 4, 4]
  <type 'instancemethod'>: 65 != 66
  <type 'gevent.libev.corecext.io'>: 8 != 10
  <class 'gevent._socket2.socket'>: 4 != 5
```

You can use the gevent-specific function `self._close_on_teardown` to avoid this...it's also common practice in the gevent test suite to manually close the resource before the test function returns.